### PR TITLE
Handle tagsAll secrets in TransformOutputs

### PR DESCRIPTION
**Pull request by iwahbe**
`tagsAll` is a required property, so the `additionalOutputSecrets` always makes the non-missing value a secret. By moving the secreting logic into `TransformOutputs`, we can keep the common case known minimal relaxation of our current assumptions.

After this PR, we still guarantee:
- If `tags` contains a secret value, then `tagsAll` will be secret.
- If `defaultTags` (provider level) contains a secret value, then `tagsAll` will be secret.

Under this PR, if the user sets `tags` as `secret({})`, then that information will leak into `tagsAll`. Observers can deduce that the value of `tags` must be `{}` because otherwise `tagsAll` would be secret`.

I'm comfortable with this, since it will reduce the number of encryptions and decryptions our users will need for the common case, speeding up deployments. I can't think how leaking this information will be impactful.

Note for reviewers:
The first commit are all manual code changes. The second are schema changes, and the third as SDK changes.

### CODE REVIEW

File: provider/tags.go [hunk 1]
```diff
@@ -125,3 +125,36 @@ func mergeTags(
 		return resource.NewNullProperty(), nil
 	}
 }
+
+func fixupTagsAll(ctx context.Context, m resource.PropertyMap) (resource.PropertyMap, error) {
```
Comment (VenelinMartinov) [inline provider/tags.go:129]: Non-blocking but can we unit test this?

**Review by VenelinMartinov — APPROVED** [review]

**Comment by VenelinMartinov** [comment]
>If tags contains a secret value, then tagsAll will be secret.
Is this handled somewhere else? I don't see logic for this in the PR

File: provider/tags.go (hunk 1 above)
Comment (iwahbe) [reply provider/tags.go:129]: Yes. That's what I'm working on now.

**Comment by t0yv0** [comment]
I think this is obsoleted by https://github.com/pulumi/pulumi-aws/commit/4c1ce0e0e9129cb02e8683d977cf290a4a8850fe

**Comment by iwahbe** [comment]
Yes it is.

**Pull request closed without merging**